com.unity.multiplayer.mlapi/Runtime/Messaging/RpcParams.cs

```diff
@@ -1,13 +1,34 @@
 namespace MLAPI.Messaging
 {
-    public struct ServerRpcSendParams
+    public interface IHasUpdateStage
     {
-        public NetworkUpdateStage UpdateStage;
+        public NetworkUpdateStage UpdateStage
+        {
+            get;
+            set;
+        }
     }
 
-    public struct ServerRpcReceiveParams
+    public struct ServerRpcSendParams: IHasUpdateStage
     {
-        public NetworkUpdateStage UpdateStage;
+        private NetworkUpdateStage m_UpdateStage;
+
+        public NetworkUpdateStage UpdateStage
+        {
+            get => m_UpdateStage;
+            set => m_UpdateStage = value;
+        }
+    }
+
+    public struct ServerRpcReceiveParams: IHasUpdateStage
```

Comment:
What if?
```suggestion
    public struct ServerRpcReceiveParams: ServerRpcSendParams
```

Reply:
Hm... in terms of the fields they have, you're right, this would be simpler. 

My first (extremely pedantic) response is that on a communication level (i.e., what we're saying about this class to our users), inheritance implies an "is-a" relationship between the two classes that I don't feel is accurate. It feels kind of like `class Car : Bicycle` in order to avoid reimplementing `public Wheel[] wheels;`.

My second response after a bit more thought is that my pedantic response is irrelevant anyway because C# doesn't allow structs to inherit from other structs (struct polymorphism isn't allowed and structs don't have a method table to support virtual method calls), so implementing the interface and duplicating the property implementation is actually the only option without introducing allocations.

Reply:
Well I learned something today!

I think it's actually they weren't as tolerant of the slicing aberration in C/C++ tho.